controllers/component_build_controller_pac.go

```diff
@@ -887,18 +893,31 @@ func createWorkspaceBinding(pipelineWorkspaces []tektonapi.PipelineWorkspaceDecl
 }
 
 // retrievePipelineSpec retrieves pipeline definition with given name from the given bundle.
-func retrievePipelineSpec(bundleUri, pipelineName string) (*tektonapi.PipelineSpec, error) {
+func retrievePipelineSpec(ctx context.Context, bundleUri, pipelineName string) (*tektonapi.PipelineSpec, error) {
+	log := ctrllog.FromContext(ctx)
+
 	var obj runtime.Object
 	var err error
 	resolver := oci.NewResolver(bundleUri, authn.DefaultKeychain)
 
-	if obj, _, err = resolver.Get(context.TODO(), "pipeline", pipelineName); err != nil {
+	if obj, _, err = resolver.Get(ctx, "pipeline", pipelineName); err != nil {
 		return nil, err
 	}
-	pipelineSpecObj, ok := obj.(tektonapi.PipelineObject)
-	if !ok {
+
+	var pipelineSpec tektonapi.PipelineSpec
+
+	if v1beta1Pipeline, ok := obj.(tektonapi_v1beta1.PipelineObject); ok {
+		v1beta1PipelineSpec := v1beta1Pipeline.PipelineSpec()
+		log.Info("Converting from v1beta1 to v1", "PipelineName", pipelineName, "Bundle", bundleUri)
+		err := v1beta1PipelineSpec.ConvertTo(ctx, &pipelineSpec)
+		if err != nil {
+			return nil, fmt.Errorf("pipeline %s from bundle %s: failed to convert from v1beta1 to v1: %w", bundleUri, pipelineName, err)
```

Comment:
Looks like a permanent error.

Reply:
Added a new permanent error for this, PTAL at the message and error ID (not sure what rules those should follow)